docs/sphinx/api/languages/cpp_api.rst

```diff
@@ -1,6 +1,46 @@
 CUDA-Q C++ API
 ******************************
 
+Operators 
+=============
+
+.. doxygenclass:: cudaq::scalar_callback
+    :members:
+
+.. doxygenclass:: cudaq::scalar_operator
+    :members:
+
+.. doxygenstruct:: cudaq::commutation_relations
+
+.. doxygenclass:: cudaq::matrix_callback
+
+.. doxygenclass:: cudaq::boson_handler
+
+.. doxygenclass:: cudaq::fermion_handler
+
+.. doxygenclass:: cudaq::spin_op
```

Comment:
Put this below sum_op, so all the handler classes a listed in successive order. 
Also add spin_op_term, boson_op, boson_op_term, fermion_op, fermion_op_term, matrix_op, matrix_op_term.